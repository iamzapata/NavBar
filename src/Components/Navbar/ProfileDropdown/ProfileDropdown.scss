@import "styles/variables.scss";

.ProfileDropdown {
  .dropdown-trigger {
    height: 100%;
  }

  .dropdown-trigger button {
    border-radius: 2rem;
    height: 100%;
    padding: 0;
  }

  img.image {
    max-height: 2.5rem;
    margin: 0 10px 0 0;
  }

  .button .icon.is-small {
    margin: 0 10px 0 0 !important;
    i {
      color: $color-gray-text;
    }
  }

  .dropdown-menu {
    left: calc(-65%);
  }
}

@media only screen and (max-width: 769px) {
  .ProfileDropdown.dropdown.navbar-item {
    display: inline-block !important;
  }
}
