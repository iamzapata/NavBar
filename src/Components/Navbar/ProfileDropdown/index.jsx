import React, { PureComponent } from "react"
import userImage from "img/user@2x.png"
import "./ProfileDropdown.scss"

class ProfileDropdown extends PureComponent {
  render() {
    return (
      <div className="ProfileDropdown dropdown navbar-item">
        <div className="dropdown-trigger">
          <button
            className="button"
            aria-haspopup="true"
            aria-controls="ProfileDropdown"
          >
            <img className="image" src={userImage} />
            <span className="icon is-small">
              <i className="fa fa-caret-down" aria-hidden="true" />
            </span>
          </button>
        </div>
        <div className="dropdown-menu" id="ProfileDropdown" role="menu">
          <div className="dropdown-content">
            <a href="#" className="dropdown-item">
              Settings
            </a>
            <a href="#" className="dropdown-item">
              Sign Out
            </a>
          </div>
        </div>
      </div>
    )
  }
}

export default ProfileDropdown
