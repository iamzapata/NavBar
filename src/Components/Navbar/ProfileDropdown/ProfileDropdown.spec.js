import React from "react"
import { render, cleanup } from "react-testing-library"
import ProfileDropdown from "./index"

describe("<ProfileDropdown />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<ProfileDropdown />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
