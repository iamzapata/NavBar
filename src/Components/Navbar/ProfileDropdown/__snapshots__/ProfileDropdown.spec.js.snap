// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileDropdown /> Should render self 1`] = `
<div
  class="ProfileDropdown dropdown navbar-item"
>
  <div
    class="dropdown-trigger"
  >
    <button
      aria-controls="ProfileDropdown"
      aria-haspopup="true"
      class="button"
    >
      <img
        class="image"
        src="test-file-stub"
      />
      <span
        class="icon is-small"
      >
        <i
          aria-hidden="true"
          class="fa fa-caret-down"
        />
      </span>
    </button>
  </div>
  <div
    class="dropdown-menu"
    id="ProfileDropdown"
    role="menu"
  >
    <div
      class="dropdown-content"
    >
      <a
        class="dropdown-item"
        href="#"
      >
        Settings
      </a>
      <a
        class="dropdown-item"
        href="#"
      >
        Sign Out
      </a>
    </div>
  </div>
</div>
`;
